cerbero/packages/distzip.py

```diff
@@ -0,0 +1,95 @@
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Library General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+import os
+import zipfile
+
+import cerbero.utils.messages as m
+from cerbero.utils import _
+from cerbero.errors import UsageError, EmptyPackageError
+from cerbero.packages import PackagerBase, PackageType
+
+
+class DistZip(PackagerBase):
+    ''' Creates a distribution zip '''
+
+    def __init__(self, config, package, store):
+        PackagerBase.__init__(self, config, package, store)
+        self.package = package
+        self.prefix = config.prefix
+        self.package_prefix = ''
+        if self.config.packages_prefix is not None:
+            self.package_prefix = '%s-' % self.config.packages_prefix
+
+    def pack(self, output_dir, devel=True, force=False, keep_temp=False,
+             split=True, package_prefix=''):
+        try:
+            dist_files = self.files_list(PackageType.RUNTIME, force)
+        except EmptyPackageError:
+            m.warning(_("The runtime package is empty"))
+            dist_files = []
+
+        if devel:
+            try:
+                devel_files = self.files_list(PackageType.DEVEL, force)
+            except EmptyPackageError:
+                m.warning(_("The development package is empty"))
+                devel_files = []
+        else:
+            devel_files = []
+
+        if not split:
+            dist_files += devel_files
+
+        if not dist_files and not devel_files:
+            raise EmptyPackageError(self.package.name)
+
+        filenames = []
+        if dist_files:
+            runtime = self._create_zip(output_dir, PackageType.RUNTIME,
+                                           dist_files, force, package_prefix)
+            filenames.append(runtime)
+
+        if split and devel and len(devel_files) != 0:
+            devel = self._create_zip(output_dir, PackageType.DEVEL,
+                                         devel_files, force, package_prefix)
+            filenames.append(devel)
+        return filenames
+
+    def _get_name(self, package_type, ext='zip'):
+        return "%s%s-%s-%s-%s%s.%s" % (self.package_prefix, self.package.name,
+                self.config.target_platform, self.config.target_arch,
+                self.package.version, package_type, ext)
+
+    def _create_zip(self, output_dir, package_type, files, force,
```

Comment:
Instead of creating a new class that duplicates all the logic and implements _create_zip, you could probably have a single DistArchive that can be configured in the constructor to use either zip or gzip (or whatever other archive compression format we need inthe future)